src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -626,7 +631,10 @@ public ReanalyzeScope(IEnumerable<ProjectId> projectIds = null, IEnumerable<Docu
 
             public string GetLanguages(Solution solution)
             {
-                Contract.ThrowIfFalse(_solutionId == null || solution.Id == _solutionId);
+                if (_solutionId != null && solution.Id != _solutionId)
+                {
+                    return string.Empty;
```

Comment:
❗️ This is awkward, since it's now impossible to differentiate between success and failure cases. If the goal is avoiding an exception, update the method to `TryGetLanguages` that returns `bool` indicating success.

Reply:
this is used for telemetry. returning bool doesn't make anything better.

Reply:
Probably add a comment/doc comment to the method mentioning used only for telemetry?

Reply:
> returning bool doesn't make anything better.

It makes it easier to understand:

1. It reveals that the API doesn't always return the value suggested by the method name
2. In the event the return value differs from the true value suggested by the method name, it reveals what the caller does with the incorrect or misleading information


If we stay with the current method names, we shoud go back to using `Contract.ThrowIfFalse`.

Reply:
@sharwell are you going to keep block it?

Reply:
➡️ Change implemented in #27915